samcli/lib/sync/infra_sync_executor.py

```diff
@@ -0,0 +1,404 @@
+        * ContentUri property of AWS::Serverless::LayerVersion
+        * Content property of AWS::Lambda::LayerVersion
+        * DefinitionBody property of AWS::Serverless::Api
+        * BodyS3Location property of AWS::ApiGateway::RestApi
+        * DefinitionUri property of AWS::Serverless::HttpApi
+        * BodyS3Location property of AWS::ApiGatewayV2::Api
+        * DefinitionUri property of AWS::Serverless::StateMachine
+        * DefinitionS3Location property of AWS::StepFunctions::StateMachine
+
+        Fields skipped during template comparison because we have recursive compare logic for nested stack:
+        * Location property of AWS::Serverless::Application
+        * TemplateURL property of AWS::CloudFormation::Stack
+
+        Fields skipped during template comparison because it's a metadata generated by SAM
+        * SamResourceId in Metadata property of all resources
+
+        Parameters
+        ----------
+        template_dict: Dict
+            The unprocessed template dictionary
+        linked_resources: List[str]
+            The corresponding resources in the other template that got processed
+        built_template_dict: Optional[Dict]
+            The built template dict that the paths didn't get replaced with packaged links yet
+
+        Returns
+        -------
+        List[str]
+            The list of resource IDs that got changed during sanitization
+        """
+
+        resources = template_dict.get("Resources", {})
+        processed_resources: Set[str] = set()
+
+        for resource_logical_id in resources:
+            resource_dict = resources.get(resource_logical_id, {})
+
+            # Built resource dict helps with determining if a field is a local path
+            if built_template_dict:
+                built_resource_dict = built_template_dict.get("Resources", {}).get(resource_logical_id, {})
+
+            resource_type = resource_dict.get("Type")
+
+            if resource_type in CODE_SYNCABLE_RESOURCES or resource_type in SYNCABLE_STACK_RESOURCES:
+                processed_resource = self._remove_resource_field(
+                    resource_logical_id,
+                    resource_type,
+                    resource_dict,
+                    linked_resources,
+                    built_resource_dict if built_template_dict else None,
+                )
+
+                if processed_resource:
+                    LOG.debug("Sanitized %s resource %s", resource_type, resource_logical_id)
+                    processed_resources.add(processed_resource)
+
+            # Remove SamResourceId metadata since this metadata does not affect any cloud behaviour
+            resource_dict.get("Metadata", {}).pop("SamResourceId", None)
+            if not resource_dict.get("Metadata"):
+                resource_dict.pop("Metadata", None)
+            LOG.debug("Sanitizing the Metadata for resource %s", resource_logical_id)
+
+        return cast(Set, sorted(processed_resources))
```

Comment:
Is this `cast` redundant?

Reply:
No, sorted method makes the typing to recognize as List